lib/index.js

```diff
@@ -2237,6 +2237,53 @@ addMethod('flatMap', function (f) {
     return this.map(f).sequence();
 });
 
+/**
+ * Creates a new Stream of values by applying each item in a Stream to each
+ * value in a Stream of functions
+ *
+ * @id ap
+ * @section Higher-order Streams
+ * @name Stream.ap(m)
+ * @param {Stream} m - incoming stream of function(s) to apply to value(s) in stream
+ * @api public
+ *
+ * var asyncUnit = () => h.of(h.nil);
```

Comment:
`h.of(h.nil)` doesn't return an empty stream. It returns a stream with a single element: `h.nil`. You want `h([])` instead.

Reply:
Correction. I think in the current implementation, it's the case that `h.of(h.nil)` will be treated as an empty stream. Semantically, though, `h.of` should always create a on-element stream, so this is more of a corner-case (possibly bug) than anything. `h([])` is still the more correct thing to use.